pennylane/templates/subroutines/approx_time_evolution.py

```diff
@@ -0,0 +1,149 @@
+
+    to implement an approximation of the time-evolution operator as:
+
+    .. math:: U \ \approx \ \displaystyle\prod_{k \ = \ 1}^{N} \displaystyle\prod_{n} e^{-i c_n H_n t / N}
+
+    with equality occurring as :math:`N \ \rightarrow \ \infty`.
+
+    .. note::
+
+        This template uses the ``PauliRot`` operation in order to implement
+        exponentiated terms of the inputted Hamiltonian. This operation only takes
+        terms that are explicitly written in terms of products of Pauli matrices (``PauliX``, ``PauliY``, ``PauliZ``, and ``Identity``).
+        Thus, each term in the Hamiltonian must be expressed this way upon input, or else an error will be raised.
+
+    Args:
+        hamiltonian (pennylane.Hamiltonian): The PennyLane Hamiltonian object representing the Hamiltonian with which the
+                                            time-evolution operator is defined. The Hamiltonian must be explicitly written
+                                            in terms of products of Pauli gates (X, Y, Z, and I).
+
+        time (int or float): The time to which the time-evolution unitary evolves a qubit register.
+
+        N (int): The number of Trotterization steps used when approximating the time-evolution operator using
+                the decomposition formula. The default value is 1.
+
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
+
+        The template is used inside a qnode:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import TimeEvolution
+
+            n_wires = 2
+            dev = qml.device('default.qubit', wires=n_wires)
+
+            coeffs = [1, 1]
+            obs = [qml.PauliX(0), qml.PauliX(1)]
+            hamiltonian = qml.Hamiltonian(coeffs, obs)
+
+            @qml.qnode(dev)
+            def circuit(time):
+                TimeEvolution(hamiltonian, time)
+                return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_wires)]
+
+        >>> circuit(1)
+        [-0.41614684 -0.41614684]
+    """
+
+    pauli = {"Identity": "I", "PauliX": "X", "PauliY": "Y", "PauliZ": "Z"}
+
+    if N is None:
+        N = 1
+
+    ###############
+    # Input checks
+
+    if not isinstance(hamiltonian, qml.vqe.vqe.Hamiltonian):
+        raise ValueError(
+            "`hamiltonian` must be of type pennylane.Hamiltonian, got {}".format(
+                type(hamiltonian).__name__
+            )
+        )
+
+    if not isinstance(N, (int, qml.variable.Variable)):
+        raise ValueError("`N` must be of type int, got {}".format(type(N).__name__))
+
+    if not isinstance(time, (int, float, qml.variable.Variable)):
+        raise ValueError("`time` must be of type float, got {}".format(type(time).__name__))
+
+    ###############
+
+    theta = []
+    pauli_words = []
+    wires = []
+
+    for i, t in enumerate(hamiltonian.ops):
```

Comment:
why `t` here? I find that confusing as it makes me think of time. Maybe `h`?

Reply:
Might be a good place to be more specific, e.g,
```python
for i, term in ...
```